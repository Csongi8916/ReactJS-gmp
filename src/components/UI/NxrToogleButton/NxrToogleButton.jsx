import React, { useState } from 'react';

import './NxrToogleButton.scss';

export default function NxrToogleButton(props) {
  return (
    <div className="NxrToogle" >
      <span className="NxrToogle_ModeTitle">{props.title}</span>
      <div className="NxrToogle_Container">
        <button value={props.firstTitle}
          className={`NxrToogle_Btn ${
            (props.shearchParam === props.firstTitle)
              || (props.sortParam === props.firstTitle)
              ? 'NxrToogle_Btn-Active'
              : 'NxrToogle_Btn-Inactive'}`}
          onClick={(e) => props.click(e)}>
          {props.firstTitle}
        </button>
        <button value={props.secondTitle}
          className={`NxrToogle_Btn ${
            (props.shearchParam === props.secondTitle)
              || (props.sortParam === props.secondTitle)
              ? 'NxrToogle_Btn-Active'
              : 'NxrToogle_Btn-Inactive'}`}
          onClick={(e) => props.click(e)}>
          {props.secondTitle}
        </button>
      </div>
    </div>
  )
}
