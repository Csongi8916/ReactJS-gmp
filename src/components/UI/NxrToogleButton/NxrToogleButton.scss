$nrRedColor: #f65261;
$nrBlackColor: #232323;
$nrGrayColor: #424242;
$nrWhiteColor: #ffffff;

.NxrToogle {
  display: flex;
  align-items: center;

  .NxrToogle_ModeTitle {
    color: #fff;
    font-size: 16px;
    margin-right: 20px;
    text-transform: uppercase;
  }

  .NxrToogle_Container {
    display: flex;
    text-align: center;
    margin-left: 10px;
    cursor: pointer;
    color: #fff;
    font-family: Montserrat, Helvetica, sans-serif;
    font-size: 16px;
    text-transform: uppercase;

    .NxrToogle_Btn {
      border: 0;
      border-radius: 3%;
      padding: 15px 25px;
    }

    .NxrToogle_Btn-Active {
      background-color: $nrRedColor;
    }

    .NxrToogle_Btn-Inactive {
      background-color: rgba( $nrGrayColor, .7 );
    }

  }
}
