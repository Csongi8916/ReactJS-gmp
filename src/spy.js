export function MySpy() {
  this.calls = 0;
}

MySpy.prototype.fn = function () {
  return () => this.calls++;
}