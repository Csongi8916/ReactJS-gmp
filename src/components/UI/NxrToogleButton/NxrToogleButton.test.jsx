import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow, mount, render } from 'enzyme';
import NxrToogleButton from './NxrToogleButton';
import { MySpy } from '../../../spy';

const values = ['test value 1', 'test value 2'];
const comp = <NxrToogleButton title="Search by" firstTitle={values[0]} secondTitle={values[1]} />
const tree = renderer.create(comp).toJSON();
const wrap = shallow(comp);


describe('NxrButton Snapshpt tests:', () => {
  test('Should render correctly', () => {
    expect(tree).toMatchSnapshot();
    tree.children[1].children[0].props.onClick();
    wrap.find('.NxrToogle_Btn').at(0).simulate('click');
    expect(wrap.getElement()).toMatchSnapshot();
  });
});

describe('NxrButton UI tests:', () => {
  it('Should active style for selected toggle', () => {
    let button = wrap.find('.NxrToogle_Btn').at(0);
    expect(button.hasClass('NxrToogle_Btn-Active')).toBe(true);
    expect(button.hasClass('NxrToogle_Btn-Inactive')).toBe(false);

    wrap.setProps({ firstTitle: values[1], secondTitle: values[0] });
    button = wrap.find('.NxrToogle_Btn').at(0);
    expect(button.hasClass('NxrToogle_Btn-Active')).toBe(false);
    expect(button.hasClass('NxrToogle_Btn-Inactive')).toBe(true);

    wrap.setProps({ firstTitle: values[0], secondTitle: values[1] });
    button = wrap.find('.NxrToogle_Btn').at(0);
    expect(button.hasClass('NxrToogle_Btn-Active')).toBe(true);
    expect(button.hasClass('NxrToogle_Btn-Inactive')).toBe(false);
  });

  it('should call handle toogle when button is clicked', () => {
    wrap.find('.NxrToogle_Btn').at(1).simulate('click');
    expect(wrap.find('.NxrToogle_Btn').at(1).hasClass('NxrToogle_Btn-Active')).toBe(true);
  });
});


