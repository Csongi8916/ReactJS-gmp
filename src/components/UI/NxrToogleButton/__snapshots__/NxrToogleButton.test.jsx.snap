// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NxrButton Snapshpt tests: Should render correctly 1`] = `
<div
  className="NxrToogle"
>
  <span
    className="NxrToogle_ModeTitle"
  >
    Search by
  </span>
  <div
    className="NxrToogle_Container"
  >
    <button
      className="NxrToogle_Btn NxrToogle_Btn-Inactive"
      onClick={[Function]}
      value="test value 1"
    >
      test value 1
    </button>
    <button
      className="NxrToogle_Btn NxrToogle_Btn-Inactive"
      onClick={[Function]}
      value="test value 2"
    >
      test value 2
    </button>
  </div>
</div>
`;

exports[`NxrButton Snapshpt tests: Should render correctly 2`] = `
<div
  className="NxrToogle"
>
  <span
    className="NxrToogle_ModeTitle"
  >
    Search by
  </span>
  <div
    className="NxrToogle_Container"
  >
    <button
      className="NxrToogle_Btn NxrToogle_Btn-Active"
      onClick={[Function]}
    >
      test value 1
    </button>
    <button
      className="NxrToogle_Btn NxrToogle_Btn-Inactive"
      onClick={[Function]}
    >
      test value 2
    </button>
  </div>
</div>
`;
